mesonbuild/compilers/compilers.py

```diff
@@ -1386,7 +1388,7 @@ def get_compileropt_value(self,
         if target:
             return env.coredata.get_option_for_target(target, key)
         else:
-            return env.coredata.get_option_for_subproject(key, subproject)
+            return env.coredata.optstore.get_value_for(key.evolve(subproject=subproject))
```

Comment:
Somewhat inconsistent that this is evolving the key here, whereas all the other similar cases chose to evolve it after form_compileropt_key. Although in this case, it avoids modifying the `if target` case.

Reply:
I have a couple of thoughts here:
 1. these are very common options, so they'll very likely hit the `OptionKey` cache very quickly
 2. This is made more difficult by some of the first wave of option refactor work that
 3. This pattern repeats everywhere, and in my WIP branch of even more cleanups this pattern is replaced by a new optstore function that cleans this up more.

Reply:
fair, especially point 3. :)